rain_api_core/urs_util.py

```diff
@@ -1,23 +1,23 @@
-
+import json
 import logging
 import os
 import urllib
 from time import time
-from json import loads
-from rain_api_core.general_util import log_context, return_timing_object, duration
-from rain_api_core.view_util import make_set_cookie_headers_jwt, get_exp_time, JWT_COOKIE_NAME
-from rain_api_core.aws_util import retrieve_secret
 
+from rain_api_core.aws_util import retrieve_secret
+from rain_api_core.general_util import duration, log_context, return_timing_object
+from rain_api_core.view_util import JWT_COOKIE_NAME, get_exp_time, make_set_cookie_headers_jwt
 
 log = logging.getLogger(__name__)
 
 
 def get_base_url(ctxt=False):
     # Make a redirect url using optional custom domain_name, otherwise use raw domain/stage provided by API Gateway.
     try:
-        return 'https://{}/'.format(
-            os.getenv('DOMAIN_NAME', '{}/{}'.format(ctxt['domainName'], ctxt['stage'])))
+        domain = os.getenv('DOMAIN_NAME') or '{}/{}'.format(ctxt['domainName'], ctxt['stage'])
+        return f'https://{domain}/'
     except (TypeError, IndexError) as e:
+        # TODO(reweeden): Should `IndexError` actually be `KeyError`?
```

Comment:
There are a few exception handlers that catch `IndexError` even though the try block seems to never be able to throw one. I suspect the author was confusing it with `KeyError` which could actually be thrown.